File: frontend/src/services/sandbox.ts
import request from '@/api/request';

export interface SandboxRequest {
  code: string;
  input_data: string;
}

export interface SandboxResponse {
  output: string;
  error: string;
  status: 'success' | 'compile_error' | 'runtime_error';
}

export const executeCode = async (payload: SandboxRequest) => {
  return request.post('/sandbox/execute', payload) as Promise<SandboxResponse>;
};

File: backend/app/schemas/sandbox.py
from pydantic import BaseModel


class SandboxRequest(BaseModel):
    code: str
    input_data: str = ""  # 标准输入


class SandboxResponse(BaseModel):
    output: str
    error: str
    status: str  # "success", "compile_error", "runtime_error"
